includes/User.php

```diff
@@ -4841,7 +4841,29 @@ protected function saveOptions() {
 			}
 		}
 
-		$dbw->delete( 'user_properties', array( 'up_user' => $this->getId() ), __METHOD__ );
+		// kinda ghetto, but :(
+		if ($wgPreferencesUseService) {
+			$regex = implode ('|', $wgUserPreferenceWhiteList['regexes']);
```

Comment:
That problem does exist, but it exists in the current, non-service implementation since a thread will forcibly delete all a user's rows in the `user_properties` table